File: chatgpt code/[].py
def generate_brackets(n, current='', open_count=0, close_count=0):
    if len(current) == 2 * n:
        print(current)
        return
    
    if open_count < n:
        generate_brackets(n, current + '(', open_count + 1, close_count)
    if close_count < open_count:
        generate_brackets(n, current + ')', open_count, close_count + 1)

n = 2
generate_brackets(n)

File: combinatorics/[].cpp
#include <iostream>
#include <string>

int generate_brackets(int a, std::string s = "", int open_count = 0, int close_count = 0){
    if(s.size() == 2 * a){
        std::cout << s << std::endl;
    }
    if(open_count < a){
        generate_brackets(a, s + '(', open_count + 1, close_count);
    }
    if(close_count < open_count){
        generate_brackets(a, s + ')', open_count, close_count + 1);
    }
    return 0;
}

int main(){
    int a;
    std::cin >> a;
    generate_brackets(a);
}